src/components/TextInput/TextInput.js

```diff
@@ -61,6 +69,14 @@ export default class TextInput extends Component {
     }
   };
 
+  onClear = () => {
```

Comment:
generally its better for class functions to use named functions:
```
//in the constructor
this.onClear = this.onClear.bind(this);

...

// here
onClear() {
  const { onClear, disabled } = this.props;
  if(!disabled) {
    ...
  }
}
```

the main benefit is stack traces having function names instead of `anonymous function` all the way down